metals/src/main/scala/scala/meta/metals/MetalsLanguageServer.scala

```diff
@@ -236,13 +235,15 @@ class MetalsLanguageServer(
       Future.unit
     }
   }.recover { case NonFatal(e) =>
-    scribe.error("Unexpected error initializing server", e)
+    scribe.error("Unexpected error initializing server: ", e)
   }.asJava
 
   override def shutdown(): CompletableFuture[Unit] = serverState.get match {
     case ServerState.Initialized(server) =>
       scribe.info("Shutting down server")
-      server.shutdown()
+      server
+        .shutdown()
+        .thenApply(_ => serverState.set(ServerState.ShuttingDown(server)))
```

Comment:
The problem was that `ServerState.ShuttingDown` was never set, so exit method always hit second, no-op case. Thanks for https://github.com/scalameta/metals/pull/4776#discussion_r1059369477 it was easy to spot the issue